components/dashboard/src/projects/NewProject.tsx

```diff
@@ -55,20 +57,34 @@ export default function NewProject() {
 
     useEffect(() => {
         if (user && authProviders && selectedProviderHost === undefined) {
+            const storedSelectedProviderHost = localStorage.getItem(SELECTED_PROVIDER_HOST_KEY);
+
+            const authProvider = authProviders.find((p) => p.host === storedSelectedProviderHost);
+
+            // if the stored provider exists and is in one of the user's identities, use it
+            if (authProvider && user.identities.some((idt) => idt.authProviderId === authProvider.authProviderId)) {
```

Comment:
Minor suggestion: It might be helpful to be extra safe and prevent bad situations like `storedSelectedProviderHost` being `null` / `undefined` / `""` combined with some `authProvider` also having `.host` be the same value.

```suggestion
            if (storedSelectedProviderHost && authProvider && user.identities.some((idt) => idt.authProviderId === authProvider.authProviderId)) {
```